integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/createTable/TestCreateTableAsSelect.scala

```diff
@@ -170,10 +175,277 @@ class TestCreateTableAsSelect extends QueryTest with BeforeAndAfterAll {
+      addProperty(CarbonCommonConstants.ENABLE_AUTO_LOAD_MERGE,
+        CarbonCommonConstants.DEFAULT_ENABLE_AUTO_LOAD_MERGE)
+  }
+
+  test("test create table as select with filter, <, and, >=") {
+    sql("DROP TABLE IF EXISTS target_table")
+    sql("DROP TABLE IF EXISTS source_table")
+    // create carbon table and insert data
+    sql(
+      """
+        | CREATE TABLE source_table(
+        |     id INT,
+        |     name STRING,
+        |     city STRING,
+        |     age INT)
+        |     STORED BY 'carbondata'
+        |     """.stripMargin)
+    sql("INSERT INTO source_table SELECT 1,'bob','shenzhen',27")
+    sql("INSERT INTO source_table SELECT 2,'david','shenzhen',31")
+    sql("INSERT INTO source_table SELECT 3,'jack','shenzhen',5")
+    sql("INSERT INTO source_table SELECT 4,'alice','shenzhen',35")
+    sql(
+      """
+        | CREATE TABLE target_table
+        | STORED BY 'carbondata'
+        | AS
+        |   SELECT city,avg(age)
+        |   FROM source_table where age > 20 and age <= 31 GROUP BY city
+      """.stripMargin)
+
+    checkAnswer(sql("SELECT * FROM target_table"), Seq(Row("shenzhen", 29)))
+  }
+
+  test("test create table as select with filter, >=, or, =") {
+    sql("DROP TABLE IF EXISTS target_table")
+    sql("DROP TABLE IF EXISTS source_table")
+    // create carbon table and insert data
+    sql(
+      """
+        | CREATE TABLE source_table(
+        |     id INT,
+        |     name STRING,
+        |     city STRING,
+        |     age INT)
+        |     STORED BY 'carbondata'
+        |     """.stripMargin)
+    sql("INSERT INTO source_table SELECT 1,'bob','shenzhen',27")
+    sql("INSERT INTO source_table SELECT 2,'david','shenzhen',31")
+    sql("INSERT INTO source_table SELECT 3,'jack','shenzhen',5")
+    sql("INSERT INTO source_table SELECT 4,'alice','shenzhen',35")
+    sql(
+      """
+        | CREATE TABLE target_table
+        | STORED BY 'carbondata'
+        | AS
+        |   SELECT city,avg(age)
+        |   FROM source_table where age >= 20 or age = 5 group by city
+      """.stripMargin)
+
+    checkAnswer(sql("SELECT * FROM target_table"), Seq(Row("shenzhen", 24.5)))
+  }
+
   override def afterAll {
     sql("DROP TABLE IF EXISTS carbon_ctas_test")
     sql("DROP TABLE IF EXISTS parquet_ctas_test")
     sql("DROP TABLE IF EXISTS orc_ctas_test")
+    sql("DROP TABLE IF EXISTS ctas_same_table_name")
+    sql("DROP TABLE IF EXISTS ctas_select_carbon")
+    sql("DROP TABLE IF EXISTS ctas_select_direct_data")
+    sql("DROP TABLE IF EXISTS ctas_select_parquet")
+    sql("DROP TABLE IF EXISTS ctas_select_orc")
+    sql("DROP TABLE IF EXISTS ctas_select_where_carbon")
+    sql("DROP TABLE IF EXISTS ctas_select_where_parquet")
+    sql("DROP TABLE IF EXISTS ctas_select_where_orc")
+    sql("DROP TABLE IF EXISTS ctas_tblproperties_test")
+    sql("DROP TABLE IF EXISTS ctas_if_table_name")
+    sql("DROP TABLE IF EXISTS source_table")
+    sql("DROP TABLE IF EXISTS target_table")
```

Comment:
@xubo245....According to me CTAS is a very small feature in terms of lines of code and internally it uses create table and insert into flow. So I think adding too many test cases for this feature  is not a good idea and it can unnecessary increase the UT test cases running time.
I believe existing test cases are sufficient to validate the basic functionality of CTAS feature. If there are any problems with the existing test cases you can make modifications and correct them but **I suggest not to add any new test cases until and unless there is some specific scenario or bug in the feature**

Reply:
I think we should add test cases for checking CTAS basic function, like filter,sum and so on. When user use CTAS feature, they usually use filter and sum etc basic function.

If no test cases to ensure the basic function, how to find the feature problem when other developer change the code logic in the future? we should ensure the function before user
encounter a problem, not fix the problem after user encounter the problem.

For UT test cases running time, I think we should consider other methods to decrease run time, such as increase the number of machine.  And we should more focus on ensure feature function,  such as increase the coverage of carbondata project, now the coverage only 71%

Reply:
@xubo245 ...I agree with your point that we should ensure feature stability. But as explained in my suggestion above CTAS internally uses Create table flow and Insert Into flow and we have enough test cases to ensure the stability of create table and insert into feature and hence we dont require to add new test cases currently for CTAS feature.
As far as coverage is concerned, existing test cases are sufficient to cover CTAS feature functionality completely. If something is not getting covered, then we need to write UT not FT.

You can identify other features were FT is not sufficient to cover all scenarios and add the required test cases.

Reply:
@manishgupta88  so your point is that : CTAS feature is reusing "create table and insert into" flows, so you proposed to add test cases from "create table and insert into" perspective , not from CTAS perspective ?  If my understanding is right,  i agree with manish's point.

Reply:
@chenliang613....your understanding is correct. As per the hierarchy below
CTAS  --> CREATE TABLE AND INSERT INTO (CTAS uses create table and insert into flow)
INSERT INTO --> SELECT AND DATA LOAD FLOW (Insert into uses select and data load flow)

I believe For Create table, Insert into, Select and Data load we have sufficient number of test cases as FT (which verify end to end functionality).  So new test cases should be added only for inter feature impact testing if required, not for direct feature testing.